### DISCUSSION TITLE
How to add JPG image sharpening?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#823] [Ask a question] [answered]
Hello!

This is no question. Perhaps a experience report or a very small tutorial. :)

If you decrease images, the images going unsharp because some pixel gone during this process. If needed, it is possible to sharp the image after decreasing the image. Usually you use the unsharp mask filter. 

On my present project I need sharp images. So I decide to modify my `image.php`. I use the php function  `imageconvolution()` to sharp the jpgs. 

But beware, the image filesize will increase. And if you see a glow around objects in the image, the image is sharpen to much.

Some more information are available here:
https://www.phpied.com/canvas-pixels-2-convolution-matrix/

I found a good adjustment for my project. 
I insert the following code in my `image.php`.

```
// Handle media file changes
    ...
       $image = $this->resizeImage($image, $widthInput, $heightInput, $widthOutput, $heightOutput);
       $image = $this->orientImage($image, $orientation);
       if ($type=="jpg") {
          $image = $this->sharpenImage($image);
       }
       if (!$this->saveImage($image, $fileName, $type, $this->yellow->system->get("imageUploadJpgQuality"))) {
           $file->error(500, "Can't write file '$fileName'!");
    ...
```
and
```
    // sharpen image
    public function sharpenImage($image) {
        $sharpen = array([0, -2, 0], [-2, 11, -2], [0, -2, 0]);
        imageconvolution($image, $sharpen, 3, 0);
        return $image;
    }
```

Perhaps it is interesting for someone else as well.

Best regards :)

**Comment by annaesvensson** [comment, marked as answer]
Thank you. 👍 Marking as answered/solved. See also #1032.
